lib/install/management/delete.go

```diff
@@ -46,10 +46,28 @@ func (d *Dispatcher) DeleteVCH(conf *config.VirtualContainerHostConfigSpec) erro
 		return nil
 	}
 
-	if err = d.DeleteVCHInstances(vmm, conf); err != nil {
-		// if container delete failed, do not remove anything else
-		log.Infof("Specify --force to force delete")
-		return err
+	d.parentResourcepool, err = d.getComputeResource(vmm, conf)
+	if err != nil {
+		log.Error(err)
+		if !d.force {
+			log.Infof("Specify --force to force delete")
+			return err
+		}
+		// Assume the cVMs and RP have already been deleted
+		log.Warnf("No container VMs found, but proceeding with delete of VCH due to --force")
```

Comment:
I do not see a path by which both `err` and `d.parentResourcepool` are non-nil so this logic will simply skip the attempt to delete the VCH when we hit the `d.parentResourcePool != nil` condition.

The text about containerVMs is also misleading given we've not looked for containerVMs at this point.